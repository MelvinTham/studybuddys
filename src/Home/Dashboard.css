@import url(http://weloveiconfonts.com/api/?family=entypo);
@import url(https://fonts.googleapis.com/css?family=Muli);
@import url('https://fonts.googleapis.com/css?family=Nunito');

html, body {
  height: 100%;
  background: url('./yellow.jpg');
  background-size: cover;
  background-attachment: fixed;

}

#sideNav {
  z-index: 1;
  width: 200px;
  height: 100vh;
  overflow-y: hidden;
  border: 2px solid grey;
  float: left;
  padding: 0;
  margin: 0;
  opacity: 0.9;
  background-color: rgb(222, 225, 229);
}

#page-content-wrapper{
  width: 100%;
  position: absolute;
  padding: 15px;
  
}

#wrapper.menuDisplayed #sidebar-wrapper{
  width: 250px;
}

#wrapper.menuDisplayed #page-content-wrapper{
  padding-left: 250px;
}

.navItems {
  margin: 0 auto;
  float: left;
  width: 200px;
  display: block;
  font-size: 25px;
  padding: 15px 0;
  text-decoration: none !important;
  color: white;

}

.navItems:hover {
  background-color: rgb(179, 179, 179);
}

.dashSec {
  font-size: 50px;
  font-family: 'Nunito', sans-serif;
  margin: 0 auto;
  padding-top: 50px;
  display:flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  flex-direction:column;
}

#titleContainer {
  margin:0;
  padding:0;
  background-color:#f0f0f0;
  font-family:Muli, sans-serif;
  color:#444;
}

.title {
  float: right;
	position: relative;
	display: block;
}
.flexCenter{
  display:flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  flex-direction:column;
}

.flexRow{
  display:flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  flex-direction:row; 
  font-family: 'Inconsolata', monospace;
  margin-left:7%;
}

.nav {  
  background-color:black;
  width: 100vw; 
  z-index: 1 !important;
  color:white;
}

/* .navbar1{
  justify-content: flex-end;
} */

.nav_b {
  display: inline-block;
  font-size: 23px;
  width: 100%;
}