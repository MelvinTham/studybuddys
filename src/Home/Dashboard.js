import React, { Component } from 'react';
import './Dashboard.css';
import Button from '@material-ui/core/Button';
import * as firebase from 'firebase';
import { Link } from 'react-router-dom';
import axios from 'axios'; // import axios library
import Courses from "./Courses";
import Navbar from "./Navbar";



class Dashboard extends Component {
    constructor(props) {
        super(props);
        this.state = {
            loggedIn:'true', // This is no longer necessary
        }
        this.logout = this.logout.bind(this);
        this.checkIfUser = this.checkIfUser.bind(this);
        this.getUserName = this.getUserName.bind(this);
    }

    logout(e){
        e.preventDefault();
        axios.get('http://localhost:3001/api/logout');
        this.props.history.push('/');
    }

    componentDidMount(){
        this.checkIfUser();
        this.getUserName();
    }

    getUserName(e){
        axios.get('http://localhost:3001/api/getUsername')
        .then(response => {
            this.setState({name : response.data.name})
        })
    }

    checkIfUser(e){
        axios.get('http://localhost:3001/api/checkLoggedIn')
        .then(response => {
            if(!(response.data.loggedIn)){
                this.props.history.push('/');
            }
        })
    }



  render() {
    return (
      <div style = {{height: "100vh", backgroundImage: "linear-gradient(to right top, #e00a0a, #e44900, #e66b00, #e58800, #e4a300)"}}>  
        <div style = {{float: "right", display: "inline-block"}}>
            <span>{this.state.name}</span>
          <Button onClick={this.logout}>Logout</Button>  
        </div> 
       
       <Navbar/>


        <h1 
            className = "dashSec"
        >
            My Courses
        </h1>

        < Courses / >
        < Courses / >
        < Courses / >
        < Courses / >
        < Courses / >
        < Courses / >
        < Courses / >
        < Courses / >
      </div>
      
     
    )
  }
}

export default Dashboard;
