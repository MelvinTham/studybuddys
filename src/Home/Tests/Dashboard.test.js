import Enzyme, { render, shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import React from 'react';
import Dashboard from './../Dashboard';
import { BrowserRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new enzymeAdapterReact16() });
const getDefaultProps = () => ({});
describe('Dashboard component', () => {
  it('shallow renders without crashing', () => {
    const {} = getDefaultProps();
    shallow(
        <Router>
            <Dashboard />
        </Router>);
  });
  it('render snapshot', () => {
    const {} = getDefaultProps();
    const wrapper = render(
        <Router>
            <Dashboard />
        </Router>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
