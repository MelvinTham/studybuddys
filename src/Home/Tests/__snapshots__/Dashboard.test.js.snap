// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component render snapshot 1`] = `
<div
  class="browserTitle"
  data-aos="fade-down"
  data-aos-duration="500"
  data-aos-easing="linear"
>
  <h1
    class="dashSec"
  >
     My Courses 
  </h1>
  <div
    class="flexCenter"
    style="margin-top:50px"
  >
    <h2>
      <br />
      <br />
      <br />
      <br />
      You Haven't Selected a Class Yet! 
      <br />
      <br />
       Select a Class by Clicking on the Left Tab 
    </h2>
  </div>
</div>
`;
